docs/preflight/v1beta3-guide.md

```diff
@@ -0,0 +1,487 @@
+- **Use `checkName`** to provide a stable, user-facing label for each check.
+- **Prefer `fail` for unmet hard requirements**, `warn` for soft requirements, and `pass` with a direct, affirmative message.
+- **Attach `uri`** to outcomes when helpful for remediation.
+- **Keep docString in sync** with the actual checks; avoid drift by templating values into both the docs and the analyzer.
+- **Ensure values files contain all required fields** since templates now directly use values without fallback defaults.
+
+
+### Values files: shape and examples
+
+Provide a values schema that mirrors your toggles and thresholds. Example full and minimal values are included in this repository:
+
+- `values-v1beta3-full.yaml` (all features enabled, opinionated defaults)
+- `values-v1beta3-minimal.yaml` (most features disabled, conservative thresholds)
+
+Typical structure:
+```yaml
+clusterVersion:
+  enabled: true
+  minVersion: "1.24.0"
+  recommendedVersion: "1.28.0"
+
+storageClass:
+  enabled: true
+  className: "standard"
+
+crd:
+  enabled: true
+  name: "samples.mycompany.com"
+
+containerRuntime:
+  enabled: true
+
+distribution:
+  enabled: true
+  supported: ["eks", "gke", "aks", "kubeadm"]
+  unsupported: []
+
+nodeResources:
+  count:
+    enabled: true
+    min: 1
+    recommended: 3
+  cpu:
+    enabled: true
+    min: "4"
+  memory:
+    enabled: true
+    minGi: 8
+    recommendedGi: 16
+  ephemeral:
+    enabled: true
+    minGi: 20
+    recommendedGi: 50
+
+workloads:
+  deployments:
+    enabled: true
+    namespace: "default"
+    name: "example-deploy"
+    minReady: 1
+
+databases:
+  postgres:
+    enabled: true
+    collectorName: "postgres"
+    uri: "postgres://user:pass@postgres:5432/db?sslmode=disable"
+  mysql:
+    enabled: true
+    collectorName: "mysql"
+    uri: "mysql://user:pass@tcp(mysql:3306)/db"
+```
+
+
+### Render, run, and extract docs
+
+You can render templates, run preflights with values, and extract requirement docs without running checks.
+
+- **Render a templated preflight spec** to stdout or a file:
+  ```bash
+  preflight template v1beta3.yaml \
+    --values values-base.yaml \
+    --values values-prod.yaml \
+    --set storage.className=fast-local \
+    -o rendered-preflight.yaml
+  ```
+
+- **Run preflights with values** (values and sets also work with `preflight` root command):
+  ```bash
+  preflight run rendered-preflight.yaml
+  # or run directly against the template with values
+  preflight run v1beta3.yaml --values values-prod.yaml --set cluster.minNodes=5
+  ```
+
+- **Extract only documentation** from enabled analyzers in one or more templates:
+  ```bash
+  preflight docs v1beta3.yaml other-spec.yaml \
+    --values values-prod.yaml \
+    --set kubernetes.enabled=true \
+    -o REQUIREMENTS.md
+  ```
+
+Notes:
+- Multiple `--values` files are merged in order; later files win.
+- `--set` uses Helm-style semantics for nested keys and types, applied after files.
+
+
+### Authoring checklist
```

Comment:
This checklist is bulleted and the one on the migration page is actual checkboxes if that's something we care about.